containerz/containerz.proto

```diff
@@ -417,6 +417,30 @@ message StartContainerRequest {
   RunAs run_as = 11;
 
   map<string, string> labels = 12;
+  
+  // Limits defines restrictions or limits upon this container.
+  message Limits {
+    // max_cpu specifies the maximum amount of cpu this container can use.
+    // For example a value of 0.5 means this container can use a maximum of
+    // half a cpu. This setting does not imply that the container is pinned
```

Comment:
Is this half a CPU or half a core? (I assume that this means 50% of a core that is exposed to the kernel -- which I think is referred to as a CPU in some commands, e.g., `top(1)`.) k8s uses the definition:

> 1 Hyperthread on a bare-metal Intel processor with Hyperthreading [ref](https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/)

I presume that this is going to some underlying cgroups API eventually, so the definition could come from there.

Reply:
Hmm googd point - I think we should define this as a CPU and the specific implementation for the container runtime can translate accordingly.

Reply:
I think we can just say "an OS-visible CPU (which may map to a hyperthread on an Intel processor with hyperthreading, or a CPU core".